### PR TITLE
Use NonRoot node server side

### PR DESCRIPTION
**Pull request by ariquinones**
Available here: https://5e962fa99d7c3a017336807d--awesome-albattani-7ab95f.netlify.com

### CODE REVIEW

File: src/dev/src/app/components/tree-view/tree-view-server-side-demo.ts [hunk 1]
```diff
@@ -180,6 +202,51 @@ export class TreeViewServerSideDemo {
...
+      nonRootServerSideTreeNodeData: NonRootTreeNode = NonRootTreeNode.instance;
+
+      nonRootServerSideDataRetriever: PagedTreeNodeDataRetriever<FoodNode> = {
+        hasChildNodes: (node: TreeNodeData<FoodNode>) => Promise.resolve(!!node.data.children && node.data.children.length > 0),
+        getChildNodeData: (node: TreeNodeData<FoodNode>) => {
+          // not used on Server Side
+        },
+        getPagedChildNodeData: (node: TreeNodeData<FoodNode>, pagingParams: PagingParams) => {
+          // On first load the initial node will be the non root tree node, dev must handle appropriately
+          const isEmptyRoot = node instanceof NonRootTreeNode;
```
Comment (JoMas971) [inline src/dev/src/app/components/tree-view/tree-view-server-side-demo.ts:222]: Why creating a variable if you're only using it once ? There is no need, just add you condition to the if().

File: src/dev/src/app/components/tree-view/tree-view-server-side-demo.ts [hunk 2]
```diff
@@ -239,6 +306,38 @@ export class TreeViewServerSideDemo {
     }
   };
 
+  // Non Root Server Side
+  nonRootServerSideTreeNodeData: NonRootTreeNode = NonRootTreeNode.instance;
+  nonRootServerSideDataRetriever: PagedTreeNodeDataRetriever<FoodNode> = {
+    hasChildNodes: (node: TreeNodeData<FoodNode>) => Promise.resolve(!!node.data.children && node.data.children.length > 0),
+    getChildNodeData: (node: TreeNodeData<FoodNode>) => Promise.resolve([]),
+    getPagedChildNodeData: (node: TreeNodeData<FoodNode>, pagingParams: PagingParams) => {
+      const isEmptyRoot = node instanceof NonRootTreeNode;
```
Comment (JoMas971) [inline src/dev/src/app/components/tree-view/tree-view-server-side-demo.ts:315]: Why creating a variable if you're only using it once ? There is no need, just add you condition to the if().

File: src/dev/src/app/components/tree-view/tree-view-server-side-demo.ts (hunk 1 above)
Comment (ariquinones) [reply src/dev/src/app/components/tree-view/tree-view-server-side-demo.ts]: I was doing it to be more explicit for the developer to notice the difference. Its fixed now!

File: src/dev/src/app/components/tree-view/tree-view-server-side-demo.ts (hunk 2 above)
Comment (ariquinones) [reply src/dev/src/app/components/tree-view/tree-view-server-side-demo.ts]: Fixed!

**Pull request merged**